### DISCUSSION TITLE
Why "Your Command Line Tools are too outdated." over and over.

### DISCUSSION THREAD

**Discussion by joshgoebel** [Homebrew/discussions#5656] [Everyday usage] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.0-82-gc5fed58
ORIGIN: https://github.com/Homebrew/brew
HEAD: c5fed58345664b86d0b8ac82148097c62fc0b3ab
Last commit: 6 hours ago
Core tap HEAD: 04107c679bf31ef8c0da489be1bcfb8d61295a16
Core tap last commit: 35 minutes ago
Core tap JSON: 06 Oct 08:02 UTC
Core cask tap HEAD: cd1e72a09b52db79ca1329822340ce8c03345188
Core cask tap last commit: 4 hours ago
Core cask tap JSON: 06 Oct 08:02 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ENV_HINTS: set
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.5 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.5/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.46.2 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.0-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: N/A
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: No Cask quarantine support available: unknown reason.

Warning: Your Command Line Tools are too outdated.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 16.0.


Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
```

### Description of issue

Despite following the suggested steps:

```
sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install
```

The version always installed is:

```
Apple clang version 15.0.0 (clang-1500.3.9.4)
```

Where-as for some reason homebrew (from reading source) expects (`os/mac/xcode.rb`):

```
        when "15" then "1600.0.20.10"
```

**Comment by carlocab** [comment, marked as answer]
Try following this:
```
Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 16.0.
```

**Reply by joshgoebel** [reply]
I wonder if we could update the error to say "if that fails" vs "alternatively"... the wording now makes it sound like both are equally valid choices that I should expect to succeed.

**Comment by aop** [comment]
My question regarding this is: How can the tools age like milk? Why is there not enough backwards compatibility so that you could, even for a short while, keep using the old version?

**Reply by SMillerDev** [reply]
Because there is no control over them from the Homebrew side, we simply don't know if you can keep using it. Sometimes Apple will replace or remove a tool without notice, and sometimes nothing happens.

However, at all times it is reported by a tool that has the following preface for a reason:
> Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

**Reply by carlocab** [reply]
> Why is there not enough backwards compatibility so that you could, even for a short while, keep using the old version?

The warning already takes this into account. This isn't shown the second Apple releases a new version of the CLT; it gets shown once we've determined that the version you're running isn't going to work well with the one that we use in CI.
